doc/UsersGuide/BuildingRunningTesting/ContainerQuickstart.rst

```diff
@@ -349,8 +400,300 @@ When all tasks show ``SUCCEEDED``, the experiment has completed successfully.
+
+.. code-block:: console
+
+   export IMG=/full/path/to/rocky9-ss192-gcc13.sif
+
+Optional platform-specific paths that require to be accessible by the container at runtime:
+
+.. code-block:: console
+
+   export BIND_ADD=/local   # Orion/Hercules, needed during run-time for interaction with Slurm job scheduler
+   export BIND_ADD=/var     # Gaea-c6
+
+Build executables using devbuild.sh script, in a similar way as described in  :ref:`Building Executables <BuildExecutables>`, 
+except placing binaries into the ``./bin`` directory. 
+This is the essential difference, as the default ./exec/ directory where the SRW App expects to find binaries 
+will be set up to contain wrappers for the actual binaries.
+
+.. code-block:: console
+
+   ./devbuild.sh --bin-dir=bin --platform=singularity --compiler=gnu \
+        | tee log.devbuild.sh_001
+
+When all the conda environments and binaries are successfully built, exit from the container:
+
+.. code-block:: console
+
+   exit
+
+
+Use Wrapper Scripts and Runtime Environment Files
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In addition to binaries and conda installs, successful build produces:
+
+* ``srw.sh`` — wrapper to launch tasks within the container
+* ``ufs-srw.env`` — runtime environment settings and environment variables
+
+Verify the following configuration in the ``srw.sh``:
+
+* ``img`` variable points to the correct ``.sif`` GNU container image file, absolute path
+* ``-B`` binds all host directories, required for access inside the container at runtime, including staged data locations
+
+
+Link Executables to Wrapper Scripts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The following code below is run interactively to create links to executables in ./exec directory to a wrapper script. 
+Make sure the $SRW variable is properly set, as done after downloading the UFS SRW App repository and dependencies. 
+
+.. code-block:: console
+
+   cd $SRW
+   export wrapper_script=${SRW}/srw.sh
+
+   mkdir -p exec
+   cd bin
+
+   for file in *; do
+       echo $file
+       if [[ "$file" != "build_settings.yaml" ]]; then
+           ln -s $wrapper_script ../exec/$file
+       else
+           cp -pv $file ../exec/.
+       fi
+   done
+
+
+Prepare configuration files 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A machine file ``singularity.yaml`` needs to be configured in ``$SRW/ush/machines`` directory 
+for running the UFS SRW App test case. Two available templates provided, 
+``singularity_gnu.srun.yaml`` and ``singularity_gnu.mpirun.yaml``. The first template is for launching
+the binaries on a host system with the ``srun`` command when Slurm job scheduler is available with PMI2 plugin, 
+the latter case is for launching the binarieson the host system using ``mpirun`` (some HPC systems may prohibit 
+use of mpirun). Use the template that fits your system, e.g.:
+
+.. code-block:: console
+
+   cp -v singularity_gnu.srun.yaml singularity.yaml
```

Comment:
There is no precedent to have additional examples of machine files within the `ush/machine` directory.  Users are expected to update the machine file with information pertinent to the machine that they are running on.

Rather than having additional singularity machine files, it would be better to let users know that they should set:
```
RUN_CMD_FCST: mpirun -n $nprocs
RUN_CMD_POST: mpirun -n $nprocs
RUN_CMD_PRDGEN: mpirun -n $nprocs
RUN_CMD_UTILS: mpirun -n $nprocs
```
for mpirun, and:
```
RUN_CMD_FCST: srun --mpi=pmi2 -n $nprocs
RUN_CMD_POST: srun --mpi=pmi2 -n $nprocs
RUN_CMD_PRDGEN: srun --mpi=pmi2 -n $nprocs
RUN_CMD_UTILS: srun --mpi=pmi2 -n $nprocs
```
for srun.

This will remove the need for multiple examples for singularity.

Reply:
Thank you for the comment! 
In a revised version, only single singularity.yaml file is retained. The documentation is changed to add information on how the MPI launch command is to be configured. 

It also require a slight change of order of the subchapters:
preparing a wflow_singularity modulefile is discussed first, in which GNU- and MPI host-based modules are added to be loaded. Then, depending on these modules, next subchapter mentions configuration of "singularity.yaml" file where either mpirun or srun are to be specified.